massa-grpc/src/stream/new_slot_execution_outputs.rs

```diff
@@ -400,14 +439,170 @@ fn filter_map_exec_output(
     if let Some(execution_event_filter) = &filters.execution_event_filter {
         if execution_event_filter.none.is_some() {
             exec_output.events.clear();
+        } else {
+            exec_output.events.0.retain(|event| {
+                if let Some(is_failure) = execution_event_filter.is_failure {
+                    if event.context.is_error != is_failure {
+                        return false;
+                    }
+                }
+
+                if let (Some(original_operation_ids), Some(origin_operation_id)) = (
+                    execution_event_filter.original_operation_ids.clone(),
+                    event.context.origin_operation_id,
+                ) {
+                    if !original_operation_ids.contains(&origin_operation_id) {
+                        return false;
+                    }
+                }
+
+                if let Some(caller_addresses) = execution_event_filter.caller_addresses.clone() {
+                    if !caller_addresses
+                        .into_iter()
+                        .any(|caller_address| event.context.call_stack.contains(&caller_address))
+                    {
+                        return false;
+                    };
+                }
+                //TODO to be confirmed
```

Comment:
@AurelienFT could you confirm this information ?

Reply:
No the emitter address should be the back of the callstack. But currently in the code it's inverted see : https://github.com/massalabs/massa/issues/4409 but we are not correcting it now because it could break too much projects : https://github.com/massalabs/massa/pull/4422#issuecomment-1729763356 (I'm not fully agree with this but not the subject) So I suggest that you match the wrong behaviour and use `.front()` like [here](https://github.com/massalabs/massa/blob/2b875f07a2fb9c40f11d78c5673e0cdd0f766c43/massa-execution-exports/src/event_store.rs#L80)